### PR TITLE
Ben add tools params

### PR DESCRIPTION
**Pull request by benlower**

### CODE REVIEW

**Review by mdepinet — COMMENTED** [review]
Your diff is showing tool auth too.  Can you sync before I review please?

**Comment by benlower** [comment]
there was a merge conflict. that is resolved now. you can ignore diffs that are before line 216. one is the merge conflict that now has suggestions you had made on the prior PR as well as some naming corrections i made to the auth examples.

File: docs/src/content/docs/api/tools.mdx [hunk 1]
```diff
@@ -210,6 +213,92 @@ You then pass in the key(s) in the `authTokens` property of `selectedTools` when
   </TabItem>
 </Tabs>
 
+## Tool Parameters
+Tool parameters define what gets passed in to your backend service when the tool is called. When creating a tool, parameters are defined as one of three types:
+<Steps>
+1. **Dynamic** → The model will choose which values to pass. Can be overridden (see [below](./#parameter-overrides)).
+1. **Static** → Value is known and unconditionally set when the call starts. Not exposed to or set by the model.
```
Comment (mdepinet) [inline docs/src/content/docs/api/tools.mdx:220]:
```suggestion
1. **Static** → Value is known when the tool is defined and is unconditionally set on invocations. Not exposed to or set by the model.
```

File: docs/src/content/docs/api/tools.mdx [hunk 2]
```diff
@@ -210,6 +213,92 @@ You then pass in the key(s) in the `authTokens` property of `selectedTools` when
   </TabItem>
 </Tabs>
 
+## Tool Parameters
+Tool parameters define what gets passed in to your backend service when the tool is called. When creating a tool, parameters are defined as one of three types:
+<Steps>
+1. **Dynamic** → The model will choose which values to pass. Can be overridden (see [below](./#parameter-overrides)).
```
Comment (mdepinet) [inline docs/src/content/docs/api/tools.mdx:219]:
```suggestion
1. **Dynamic** → The model will choose which values to pass. These are the parameters you'd use for a single-generation LLM API. Can be overridden (see [below](./#parameter-overrides)).
```

File: docs/src/content/docs/api/tools.mdx [hunk 3]
```diff
@@ -210,6 +213,92 @@ You then pass in the key(s) in the `authTokens` property of `selectedTools` when
...
+1. **Static** → Value is known and unconditionally set when the call starts. Not exposed to or set by the model.
+1. **Automatic** → Automatically set by the system. Not known when the call starts but known at tool invocation.
```
Comment (mdepinet) [inline docs/src/content/docs/api/tools.mdx:221]:
```suggestion
1. **Automatic** → Like "Static", except that the value may not be known when the tool is defined but will instead be populated by the system when the tool is invoked.
```

File: docs/src/content/docs/api/tools.mdx [hunk 4]
```diff
@@ -210,6 +213,92 @@ You then pass in the key(s) in the `authTokens` property of `selectedTools` when
...
+{
+  "model": "fixie-ai/ultravox-70B",
+  "systemPrompt": ...
+  "selectedTools": [
+    "toolName": "colorPicker",
+    "parameterOverrides": {
+      "color": "green"
+    }
+  ]
+}
```
Comment (mdepinet) [inline docs/src/content/docs/api/tools.mdx:261]:
```suggestion
// Always set the stock symbol to 'NVDA'
{
  "model": "fixie-ai/ultravox-70B",
  "systemPrompt": ...
  "selectedTools": [
    "toolName": "stock_price",
    "parameterOverrides": {
      "symbol": "NVDA"
    }
  ]
}
```

(Seems useful to relate this to your parameter definition above.  Up to you.)

File: docs/src/content/docs/api/tools.mdx [hunk 5]
```diff
@@ -210,6 +213,92 @@ You then pass in the key(s) in the `authTokens` property of `selectedTools` when
...
+If you have parameters that are known at the time you create the tool, static parameters can be used.
+
+```javascript
+// Adding a static parameter to a stock price tool
+{
+  "name": "stock_price",
+  "description": "Get the current stock price for a given symbol",
+  "staticParameters": [
+    {
+      "name": "close_date",
```
Comment (mdepinet) [inline docs/src/content/docs/api/tools.mdx:276]: This seems like something that ought to be an automatic parameter (not that it's something we support right now - just seems like we could have a better example).  Should this maybe be start_date?  Or maybe something more generic like adding a utm=ultravox query param?

File: docs/src/content/docs/api/tools.mdx [hunk 6]
```diff
@@ -210,6 +213,92 @@ You then pass in the key(s) in the `authTokens` property of `selectedTools` when
...
+      "location": "PARAMETER_LOCATION_QUERY",
+      "value": "2024-06-01"
+    }
+  ]
+}
+```
+
+### Automatic Parameters
+Automatic parameters are used when you don't want the model to specify the value and you don't know the value when the tool is created. The primary use case for automatic parameters today is for using the `call_id` that is generated for the current call and then passing it as a unique identifier to your tool.
```
Comment (mdepinet) [inline docs/src/content/docs/api/tools.mdx:285]: fwiw, you can also use automatic parameters to include the call's messages.  I added this just a few minutes ago to support call update tools.

File: docs/src/content/docs/api/tools.mdx (hunk 5 above)
Comment (benlower) [reply docs/src/content/docs/api/tools.mdx:276]: yeah will change to utm idea

File: docs/src/content/docs/api/tools.mdx (hunk 6 above)
Comment (benlower) [reply docs/src/content/docs/api/tools.mdx:285]: added a mention

**Review by mdepinet — APPROVED** [review]

**Pull request merged**